server/server_test.go

```diff
@@ -1,28 +1,66 @@
 package main
 
 import (
+	"fmt"
 	"net/http"
 	"net/http/httptest"
+	"strconv"
 	"strings"
 	"testing"
 
 	"github.com/matryer/is"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/goleak"
 )
 
+const (
+	exists = "exists"
+)
+
+func TestDelete(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name string
+		code int
+		path string
+		key  string
+	}{
+		{name: "key exists", code: http.StatusOK, path: "/db", key: "test"},
+		{name: "key does not exist", code: http.StatusNotFound, path: "/db", key: "test1"},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			is := is.New(t)
+
+			db := make(map[string]string)
+			db["test"] = "succeeded"
+			s := testServer(db)
+
+			req := httptest.NewRequest(http.MethodDelete, fmt.Sprintf("%s?key=%s", tt.path, tt.key), nil)
+			w := httptest.NewRecorder()
+			s.serveHTTP(w, req)
+			is.Equal(w.Code, tt.code)
+		})
+	}
+}
+
 func TestGet(t *testing.T) {
 	t.Parallel()
 	tests := []struct {
 		name     string
 		code     int
 		path     string
+		key      string
 		respBody string
 	}{
-		{name: "works, test ok", code: http.StatusOK, path: "/?key=test", respBody: "succeeded"},
-		{name: "not working, test ok", code: http.StatusNotFound, path: "/?key=not-there", respBody: "succeeded"},
-		{name: "not working, wrong value", code: http.StatusNotFound, path: "/?key=not"},
-		{name: "not working, no value", code: http.StatusNotFound, path: "/?key=", respBody: ""},
-		{name: "not working, wrong path", code: http.StatusBadRequest, path: "/test/?key=not", respBody: ""},
-		{name: "not working, wrong path 2", code: http.StatusBadRequest, path: "/test/test?key=not", respBody: ""},
+		{name: "test ok", code: http.StatusOK, path: "/db", key: "test", respBody: "succeeded"},
+		{name: "test key not found", code: http.StatusNotFound, path: "/db", key: "not-there", respBody: "succeeded"},
+		{name: "wrong value", code: http.StatusNotFound, path: "/", key: "not"},
+		{name: "no value", code: http.StatusNotFound, path: "/", key: "", respBody: ""},
```

Comment:
these two requests don't even hit get get handler

Reply:
agreed, they should be in their own test